tests/test_forms.py

```diff
@@ -5,42 +5,116 @@
 from rasa_sdk.executor import CollectingDispatcher
 from rasa_sdk.forms import FormAction
 
-from actions.actions import HealthCheckProfileForm
+from actions.actions import HealthCheckProfileForm, HealthCheckTermsForm
+from tests import utils
 
 
 class TestHealthCheckProfileForm:
-    def get_tracker_for_number_slot_with_value(self, slot_name, value):
-        return Tracker(
-            "default",
-            {"requested_slot": slot_name},
-            {"entities": [{"entity": "number", "value": value}]},
-            [],
-            False,
-            None,
-            {},
-            "action_listen",
+    @pytest.mark.asyncio
+    async def test_validate_age(self):
+        form = HealthCheckProfileForm()
+        dispatcher = CollectingDispatcher()
+
+        tracker = utils.get_tracker_for_number_slot_with_value(form, "age", "1")
+        events = await form.run(dispatcher=dispatcher, tracker=tracker, domain=None)
+        assert events == [SlotSet("age", "<18"), SlotSet("requested_slot", "gender")]
+
+        tracker = utils.get_tracker_for_number_slot_with_value(form, "age", "2")
+        events = await form.run(dispatcher=dispatcher, tracker=tracker, domain=None)
+        assert events == [SlotSet("age", "18-39"), SlotSet("requested_slot", "gender")]
+
+        tracker = utils.get_tracker_for_number_slot_with_value(form, "age", "3")
+        events = await form.run(dispatcher=dispatcher, tracker=tracker, domain=None)
+        assert events == [SlotSet("age", "40-65"), SlotSet("requested_slot", "gender")]
+
+        tracker = utils.get_tracker_for_number_slot_with_value(form, "age", "4")
+        events = await form.run(dispatcher=dispatcher, tracker=tracker, domain=None)
+        assert events == [SlotSet("age", ">65"), SlotSet("requested_slot", "gender")]
+
+        tracker = utils.get_tracker_for_number_slot_with_value(form, "age", ["2", "39"])
+        events = await form.run(dispatcher=dispatcher, tracker=tracker, domain=None)
+        assert events == [SlotSet("age", None), SlotSet("requested_slot", "age")]
+
+        tracker = utils.get_tracker_for_number_slot_with_value(
+            form, "age", "not a number"
         )
+        events = await form.run(dispatcher=dispatcher, tracker=tracker, domain=None)
+        assert events == [SlotSet("age", None), SlotSet("requested_slot", "age")]
 
     @pytest.mark.asyncio
-    async def test_validate_age(self):
+    async def test_validate_gender(self):
+        form = HealthCheckProfileForm()
+        dispatcher = CollectingDispatcher()
+
+        tracker = utils.get_tracker_for_number_slot_with_value(
+            form, "gender", "1", {"age": "18-39"}
```

Comment:
When I was testing the location, I didn't need to fill in the other slots, setting the `requested_slot` was enough. Just want to see if we've checked if this will work without having to fill all the previous slots on every test.

Reply:
@rudigiesler if you just want to validate the 1 slot then it's fine, but I noticed it wasn't setting the right value for previous slots.